### PR TITLE
Backend admin

### PR DESCRIPTION
**Pull request by vaastav**
This PR adds all the get queries to our backend in order to get all the data stored in the tables except for ContractorRecords

### CODE REVIEW

File: code/backend/src/main/java/server/rest/controllers/Controller.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package server.rest.controllers;
+
+/**
+ * Created by vaast on 30/10/2017.
+ */
+public class Controller {
```
Comment (StephenVanSon) [inline code/backend/src/main/java/server/rest/controllers/Controller.java]: I moved these to a config file with a bean defined (however haven't pushed). I'll change this myself once I pull your changes into my local branch.

File: code/backend/src/main/java/server/rest/controllers/FXRateController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import server.rest.responses.FXRatesResponse;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@CrossOrigin(origins = "http://localhost:1234")
```
Comment (StephenVanSon) [inline code/backend/src/main/java/server/rest/controllers/FXRateController.java]: Same origin thing as above. I can change myself.

File: code/backend/src/main/java/server/rest/controllers/HRPayGradeController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import server.rest.responses.HRPayGradeResponse;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@CrossOrigin(origins = "http://localhost:1234")
```
Comment (StephenVanSon) [inline code/backend/src/main/java/server/rest/controllers/HRPayGradeController.java]: See above comments.

File: code/backend/src/main/java/server/rest/controllers/HRPositionRoleController.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import server.rest.responses.HRPositionRoleResponse;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@CrossOrigin(origins = "http://localhost:1234")
```
Comment (StephenVanSon) [inline code/backend/src/main/java/server/rest/controllers/HRPositionRoleController.java]: Origin. I can change all these.

File: code/backend/src/main/java/server/rest/controllers/SkillsController.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import server.rest.responses.SkillsResponse;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@CrossOrigin(origins = "http://localhost:1234")
```
Comment (StephenVanSon) [inline code/backend/src/main/java/server/rest/controllers/SkillsController.java]: See others.

File: code/backend/src/main/java/server/model/Contractor.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    /**
+     * Creates a Contractor Object
+     * @param id Id of the contractor
+     * @param firstName First Name of the Contractor
+     * @param lastName Last Name of the Contractor
+     * @param agencySource Agency Source of the Contractor
+     * @param status Status of the Contractor
+     * @param rehire Rehire status of the Contractor
+     */
+    public Contractor(int id, String firstName, String lastName, String agencySource, String status, boolean rehire) {
```
[the remark this answers is not in the document]
Comment (vaastav) [reply code/backend/src/main/java/server/model/Contractor.java]: Planning to add that later.

File: code/backend/src/main/java/server/rest/controllers/Controller.java (hunk 1 above)
Comment (StephenVanSon) [reply code/backend/src/main/java/server/rest/controllers/Controller.java]: Another side-note, having it in config allows us to define it on the elastic beanstalk environment, so say we were running a different DB in that environment then (local vs hosted) then we wouldn't have any code changes necessary for the connection string to change.

File: code/backend/src/main/java/server/rest/controllers/SkillsController.java (hunk 5 above)
Comment (vaastav) [reply code/backend/src/main/java/server/rest/controllers/SkillsController.java]: @StephenVanSon  Can you make the changes to the origins after I merging this? I don't think I can get that done today

Comment (StephenVanSon) [reply code/backend/src/main/java/server/rest/controllers/SkillsController.java]: Yup I can

**Pull request merged**